plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java

```diff
@@ -28,17 +42,59 @@
  */
 public class GoogleLoginService {
 
-  // TODO(chanseok): this will be needed later.
-  public static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
-      "https://www.googleapis.com/auth/userinfo#email"
+  private static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
+      "email", //$NON-NLS-1$
+      "https://www.googleapis.com/auth/cloud-platform" //$NON-NLS-1$
   ));
 
   /**
    * Returns the credential of the active user. If there is no active user, returns {@code null}.
    */
   // Should probably be synchronized properly.
-  // TODO(chanseok): consider returning a String JSON (i.e., hide Credential)
-  public Credential getActiveCredential() {
-    return null;
+  public Credential getActiveCredential(IShellProvider shellProvider) throws IOException {
```

Comment:
Hmm, now this is where it gets interesting and we have to be careful.  I wouldn't suggest associating credentials with `Shell`s but with `IWorkbenchWindow`/`MWindow`.  This code would look up the `Shell` and its parent `Shell`s (in case the `Shell` is a dialog), and try to find the IWW/MW.  But rather than try to embed that map-Shell-to-MW/IWW logic here, I'd push it onto the caller and just have this method be `getAssignedCredential(IWorkbenchWindow)` and we can expand others later as required.